### DISCUSSION TITLE
Поднять на роутере Keenetic

### DISCUSSION THREAD

**Discussion by Montigomo** [hufrea/byedpi#65] [General]
Здравствуйте. Идея такая поднять этот прокси на роутере Кинетик с Entware. В самом Кинетике есть компонент прокси соединения, его замкнуть на поднятый сервер. И все соединения будут проходить через прокси на роутере. Но что то не получается. Может сориентируете какие параметры надо указать? 
Прослушиваемый IP - 127.0.0.1?
I  --conn-ip - тут тоже не пойму, этот IP что мне раздает провайдер или IP роутера в локалке?
И как считаете реализуемо такое?

**Comment by zetcamp** [comment]
По-умолчанию, если Вы не указали явно параметр "-i" с адресом, то прокси поднимается на 0.0.0.0:1080, т.е. на всех интерфейсах.

Чтобы ваш прокси был доступен в локалке (если надо), поднимите его на кинетике как-то так:
./ciadpi --disorder 1 --auto=torst --tlsrec 1+s -i 192.168.1.1 -p 1080

Это пример с параметрами из описания, где вместо 192.168.1.1 Вам следует указать свой локальный адрес роутера, если у Вас он другой.

После чего можно попробовать поднять в веб-морде кинетика прокси соединение (socks5://192.168.1.1:1080) и назначить его приоритетным.

И в консоли смотреть на выхлоп при обращении к заблокированным ресурсам - будут ли там какие-то телодвижения.

Подбирать параметры обхода нужно уже на месте для Вашего провайдера.

Если у Вас уже установлена Entware, то при успешном тестировании прокси - прописать его автозапуск, например, в /opt/etc/rc.d/ (если я правильно помню). Или как там на кинетиках ещё можно.

**Comment by azhinu** [comment]
Попробовала это сделать, вылетает ошибка: `accept: No file descriptors available`

**Comment by zetcamp** [comment]
Какой выхлоп даёт "opkg print-architecture"?

**Comment by azhinu** [comment]
@zetcamp 
arch all 100
arch mipsel-3.4 150
arch mipsel-3.4_kn 200

**Comment by zetcamp** [comment]
@azhinu с виду всё нормально.
На кн-1011 пробовал бинарник для mipsel запускать, такой ошибки не получал. Или просто не дошёл до неё.

**Comment by azhinu** [comment]
@zetcamp ошибка появляется именно при нагрузке.

**Comment by zetcamp** [comment]
@azhinu там в соседней ветке дали ссылку на бинарь другой сборки.
https://github.com/spvkgn/ByeDPI-OpenWrt/releases

**Comment by crim50n** [comment]
Хочу поделиться своими наблюдениями. У меня модель KN-1810. Бинарный файл для архитектуры mipsel работает отлично, но в отличие от 3proxy, который стоит рядом и работает без проблем, стандартный клиент прокси в Keenetic с ByeDPI не может разрешить DNS. Буду благодарен за помощь в решении этой проблемы.

UPD: Починил. Сервис запускается с параметрами `--hosts /opt/etc/ciadpi/hosts.txt --disorder 1 --auto=none --tlsrec 1+s -i 0.0.0.0 -p 36018 -x 1`, но почему-то спустя короткое время отваливается
```
host: c (111)
tlsrec: pos=107, n=2738
new conn: fd=1014, addr=1.0.0.1:853
accept: fd=1015
desync params index: 1
host: c (111)
tlsrec: pos=107, n=2743
new conn: fd=1016, addr=1.0.0.1:853
accept: fd=1017
desync params index: 1
host: c (111)
tlsrec: pos=107, n=2748
new conn: fd=1018, addr=1.0.0.1:853
accept: fd=1019
desync params index: 1
host: c (111)
tlsrec: pos=107, n=2753
new conn: fd=1020, addr=1.0.0.1:853
accept: fd=1021
desync params index: 1
host: c (111)
tlsrec: pos=107, n=2758
new conn: fd=1022, addr=1.0.0.1:853
accept: fd=1023
accept: No file descriptors available
exit
```

UPD2: Всё починил, но решил снести из-за слишком низкой производительности badvpn-tun2socks.

**Reply by kvvoff** [reply]
Приветствую! Как починили? У меня так же отваливается с тем же логом

**Comment by jura1243** [comment]
у меня получилось на домашнем сервере ciadpi, а роутер для всех как сокс прокси по этой инструкции: https://help.keenetic.com/hc/ru/articles/7474374790300-Клиент-прокси . только почемуто старые смарт ТВ отказались работать.

**Reply by jura1243** [reply]
перестала работать эта тема.

**Comment by smazilov** [comment]
Какой пакет вам удалось установить на Keenetic? На KN-1910 (arch=mipsel, cpu_type=mips32r2) пакет byedpi_0.16.5-1_mipsel_mips32.ipk не встаёт...

**Reply by zetcamp** [reply]
Вы где его вообще взяли, пакет этот?

**Reply by jura1243** [reply]
> Вы где его вообще взяли, пакет этот?

например тут все бинари:
https://github.com/hufrea/byedpi/releases/tag/v0.16.5

**Reply by smazilov** [reply]
Достаточно копнуть чуть поглубже https://github.com/spvkgn/ByeDPI-OpenWrt/releases (для OpenWrt 23.05 и более старых)
Так какой правильный-то?

**Reply by jura1243** [reply]
>  Так какой правильный-то?

спросите об этом ИИ. для моего  Keenetic Speedster наиболее вероятной архитектурой является MIPSEL (Little Endian).

**Reply by zetcamp** [reply]
> > Вы где его вообще взяли, пакет этот?
> 
> например тут все бинари: https://github.com/hufrea/byedpi/releases/tag/v0.16.5

Человек ставит именно ipk в opkg, а не просто бинарь из этогг реаозитория.

**Reply by jura1243** [reply]
> > > Вы где его вообще взяли, пакет этот?
> > 
> > 
> > например тут все бинари: https://github.com/hufrea/byedpi/releases/tag/v0.16.5
> 
> Человек ставит именно ipk в opkg, а не просто бинарь из этогг реаозитория.

у меня нет usb порта чтобы ставить свои сборки.

**Reply by zetcamp** [reply]
> Достаточно копнуть чуть поглубже https://github.com/spvkgn/ByeDPI-OpenWrt/releases (для OpenWrt 23.05 и более старых) Так какой правильный-то?

Вы обсуждаете левый пакет из другого репозитория здесь, серьёзно? Может стоит там и спрашивать?

А по теме - вам нужно аопробовать этот:
byedpi_0.16.5-r1_mipsel_24kc.ipk

**Reply by smazilov** [reply]
Человек просто не умеет устанавливать бинарь из этого репозитория и нигде никак не найдёт понятной инструкции, как это сделать...

**Reply by smazilov** [reply]
> А по теме - вам нужно аопробовать этот:
> byedpi_0.16.5-r1_mipsel_24kc.ipk

Этот пакет тоже пробовал (и перепроверил ещё раз) - не пошло...

**Comment by a4lex** [comment]
У себя запустил так:

cat /opt/etc/init.d/S90ciadpi

```
#!/bin/sh

# ciadpi init script for Keenetic
# Location: /opt/etc/init.d/S90ciadpi

DAEMON="/opt/root/ciadpi-mipsel"
PIDFILE="/opt/var/run/ciadpi.pid"
ARGS="--ip 192.168.4.1 -p 54321 -o 1:3:5 --transparent --daemon --pidfile $PIDFILE"

add_iptables_rules() {
    iptables -t nat -A PREROUTING -p tcp --dport 443 -j REDIRECT --to-port 54321
    iptables -t nat -A PREROUTING -p tcp --dport 80 -j REDIRECT --to-port 54321
    echo "Added iptables rules"
}

remove_iptables_rules() {
    iptables -t nat -D PREROUTING -p tcp --dport 443 -j REDIRECT --to-port 54321
    iptables -t nat -D PREROUTING -p tcp --dport 80 -j REDIRECT --to-port 54321
    echo "Removed iptables rules"
}

start() {
    if [ -f "$PIDFILE" ] && kill -0 $(cat "$PIDFILE") 2>/dev/null; then
        echo "ciadpi is already running"
        return 1
    fi

    echo "Starting ciadpi..."
    add_iptables_rules
    $DAEMON $ARGS
}

stop() {
    if [ ! -f "$PIDFILE" ]; then
        echo "ciadpi is not running"
        return 1
    fi

    echo "Stopping ciadpi..."
    remove_iptables_rules
    kill $(cat "$PIDFILE")
    # rm -f "$PIDFILE"
}

restart() {
    stop
    sleep 1
    start
}

case "$1" in
    start)
        start
        ;;
    stop)
        stop
        ;;
    restart)
        restart
        ;;
    *)
        echo "Usage: $0 {start|stop|restart}"
        exit 1
        ;;
esac

exit 0
```
